subprojects/instrumentation-declarations/build.gradle.kts

```diff
@@ -0,0 +1,28 @@
+plugins {
+    id("gradlebuild.distribution.api-java")
+}
+
+description = "Contains declarations for instrumentation of plugins. Adds interceptors, bytecode upgrades etc."
+
+dependencies {
+    compileOnly(project(":core"))
+    compileOnly(project(":base-services"))
+    compileOnly(project(":core-api"))
+    compileOnly(project(":model-core"))
+    compileOnly(project(":reporting"))
+    compileOnly(libs.groovy)
+    compileOnly(project(":code-quality"))
+
+    // Instrumentation dependencies
+    compileOnly(project(":internal-instrumentation-api"))
+    compileOnly(libs.asm)
+    compileOnly(libs.asmUtil)
+    compileOnly(libs.asmTree)
+    annotationProcessor(project(":internal-instrumentation-processor"))
+    annotationProcessor(platform(project(":distributions-dependencies")))
+}
+
+tasks.named<JavaCompile>("compileJava") {
+    // Without this, javac will complain about unclaimed annotations
+    options.compilerArgs.add("-Xlint:-processing")
```

Comment:
❔ Which annotations does it complain about and why?

Reply:
Annotation

Reply:
Looks like the `org.gradle.api.NonNullApi`. You get this kind of warning, that breaks the build since we have -Werror:
```
warning: No processor claimed any of these annotations: org.gradle.api.NonNullApi
error: warnings found and -Werror specified
1 error
1 warning
```

I guess we would need to try fix that at one point

Reply:
Added annotation info to the comment

Reply:
If @alllex changed the default for non-nullable APIs, we could lose the annotation from our packages, and then this problem would go away.